aptos-move/framework/aptos-framework/sources/native_bridge.move

```diff
@@ -0,0 +1,794 @@
+module aptos_framework::native_bridge {
+
+    use aptos_framework::account;
+    use aptos_framework::native_bridge_core;
+    use aptos_framework::native_bridge_configuration;
+    use aptos_framework::native_bridge_configuration::assert_is_caller_relayer;
+    use aptos_framework::native_bridge_store;
+    use aptos_std::smart_table;
+    use aptos_framework::ethereum;
+    use aptos_framework::ethereum::EthereumAddress;
+    use aptos_framework::event::{Self, EventHandle}; 
+    use aptos_framework::signer;
+    use aptos_framework::system_addresses;
+    #[test_only]
+    use aptos_framework::aptos_account;
+    #[test_only]
+    use aptos_framework::aptos_coin::AptosCoin;
+    #[test_only]
+    use aptos_framework::native_bridge_store::assert_valid_bridge_transfer_id;
+    #[test_only]
+    use aptos_framework::coin;
+    #[test_only]
+    use aptos_framework::ethereum::valid_eip55;
+    use std::bcs;
+    use std::vector;
+    use aptos_std::aptos_hash::keccak256;
+
+    const ETRANSFER_ALREADY_PROCESSED: u64 = 1;
+    const EINVALID_BRIDGE_TRANSFER_ID: u64 = 2;
+    const EEVENT_NOT_FOUND : u64 = 3;
+    const EINVALID_NONCE : u64 = 4;
+
+    #[event]
+    /// An event triggered upon initiating a bridge transfer
+    struct BridgeTransferInitiatedEvent has store, drop {
+        bridge_transfer_id: vector<u8>,
+        initiator: address,
+        recipient: vector<u8>,
+        amount: u64,
+        nonce: u64,
+    }
+
+    #[event]
+    /// An event triggered upon completing a bridge transfer
+    struct BridgeTransferCompletedEvent has store, drop {
+        bridge_transfer_id: vector<u8>,
+        initiator: vector<u8>,
+        recipient: address,
+        amount: u64,
+        nonce: u64,
+    }
+
+    /// This struct will store the event handles for bridge events.
+    struct BridgeEvents has key, store {
```

Comment:
What are event_handles? What do they store?
And why do we need to store them?
Do we use them later on?

Reply:
Aptos Move supports two forms of events: module events and EventHandle events. See https://aptos.dev/en/network/blockchain/events

Event handles are the older form, which the relayer currently uses. We could upgrade to module events for the sake of using the most modern form, but it's not technically necessary.